curriculum/challenges/english/25-front-end-development/workshop-music-instrument-filter/672137c47fa88d6f753f6d0b.md

```diff
@@ -0,0 +1,333 @@
+---
+id: 672137c47fa88d6f753f6d0b
+title: Step 12
+challengeType: 0
+dashedName: step-12
+---
+
+# --description--
+
+Currently, your `instrumentCards` function returns an array with instrument objects, so you'll need another couple of steps before you can display your instrument cards on the page.
+
+Modify your function so that it returns an array of strings containing instrument cards, each corresponding to an object in the `instruments` array.
```

Comment:
```suggestion
Modify your function so that it returns an array of strings containing the HTML code to display the instrument cards, each string corresponding to an object in the `instruments` array. The strings should have this format `<div class="card"><h2>[instrument]</h2><p>$[price]</p></div>`
```

I couldn't figure out what to do here from just the instructions. I needed to look at the hint to figure out what was expected - which wasn't too bad, but I think you should be able to solve a challenge from just the instructions. Perhaps I'm missing some context from previous workshops or lectures that would have helped here. I think this suggestion clarifies it enough.